samples/attested_tls/common/openssl_utility.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include "openssl_utility.h"
+#include <stdio.h>
+#include <string.h>
+
+oe_result_t generate_certificate_and_pkey(X509*& cert, EVP_PKEY*& pkey)
+{
+    oe_result_t result = OE_FAILURE;
+    SSL_CTX_set_ecdh_auto(ctx, 1);
+    uint8_t* output_cert = nullptr;
+    size_t output_cert_size = 0;
+    uint8_t* private_key_buf = nullptr;
+    size_t private_key_buf_size = 0;
+    uint8_t* public_key_buf = nullptr;
+    size_t public_key_buf_size = 0;
+    const unsigned char* cert_buf_ptr = nullptr;
+    BIO* mem = nullptr;
+
+    result = generate_key_pair(
+        &public_key_buf,
+        &public_key_buf_size,
+        &private_key_buf,
+        &private_key_buf_size);
+
+    // OE_CHECK_MSG(result, " failed with %s\n", oe_result_str(result));
+    printf("public_key_buf_size:[%ld]\n", public_key_buf_size);
+    printf("public key used:\n[%s]", public_key_buf);
+
+    result = oe_generate_attestation_certificate(
+        (const unsigned char*)"CN=Open Enclave SDK,O=OESDK TLS,C=US",
+        private_key_buf,
+        private_key_buf_size,
+        public_key_buf,
+        public_key_buf_size,
+        &output_cert,
+        &output_cert_size);
+    // OE_CHECK_MSG(result, " failed with %s\n", oe_result_str(result));
+
+    // temporary buffer required as if d2i_x509 call is successful cert_buf_ptr
+    // is incremented to the byte following the parsed data. sending
+    // cert_buf_ptr as argument will keep output_cert pointer undisturbed.
+    cert_buf_ptr = output_cert;
+
+    if ((cert = d2i_X509(nullptr, &cert_buf_ptr, (long)output_cert_size)) ==
+        nullptr)
+    {
+        printf("Failed to convert DER fromat certificate to X509 structure\n");
+        goto done;
+    }
+    mem = BIO_new_mem_buf((void*)private_key_buf, -1);
+    if (!mem)
+    {
+        printf("Failed to convert private key buf into BIO_mem\n");
+        goto done;
+    }
+    if ((pkey = PEM_read_bio_PrivateKey(mem, nullptr, 0, nullptr)) == nullptr)
+    {
+        printf("Failed to convert private key buffer into EVP_KEY format\n");
+        goto done;
+    }
+
+    result = OE_OK;
+done:
+    cert_buf_ptr = nullptr;
+    BIO_free(mem);
+    oe_free_key(private_key_buf, private_key_buf_size, nullptr, 0);
+    oe_free_key(public_key_buf, public_key_buf_size, nullptr, 0);
+    oe_free_attestation_certificate(output_cert);
+    return result;
+}
+
+oe_result_t load_ssl_certificates_and_keys(
```

Comment:
```suggestion
oe_result_t load_tls_certificates_and_keys(
```
Unless you're really using SSL and not TLS as the protool